### PR TITLE
delete the engine's timeoutChecker scheduler and merge it to  sessionManager's scheduler.

### PR DESCRIPTION
**Pull request by xjzrc**
delete the engine's timeoutChecker scheduler and merge it to sessionManager's scheduler.

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 1]
```diff
@@ -29,41 +29,41 @@ import org.apache.kyuubi.engine.spark.session.SparkSQLSessionManager
...
-    timeoutChecker.shutdown()
-    timeoutChecker.awaitTermination(10, TimeUnit.SECONDS)
   }
 
   override def start(): Unit = {
+    super.start()
+    startTimeoutChecker()
+  }
+
+  private def startTimeoutChecker(): Unit = {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:50]: move this check task to KyuubiSessionManager

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 2]
```diff
@@ -51,6 +51,9 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
 
   def submitBackgroundOperation(r: Runnable): Future[_] = execPool.submit(r)
 
+  def submitTimeoutChecker(r: Runnable, interval: Long, unit: TimeUnit): Future[_] =
+    timeoutChecker.scheduleWithFixedDelay(r, interval, interval, unit)
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:55]: we do need this `timeoutChecker` too, we can use `execPool` directly

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 3]
```diff
@@ -65,6 +68,8 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
     if (session == null) {
       throw KyuubiSQLException(s"Invalid $sessionHandle")
     }
+    info(s"Session with $sessionHandle is closed," +
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:71]: $sessionHandle is closed, current opening sessions ...

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala (hunk 1 above)
Comment (xjzrc) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:50]: ok, I will move it.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala (hunk 2 above)
Comment (xjzrc) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:55]: ok, I will do it.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala (hunk 3 above)
Comment (xjzrc) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala]: I will fix it.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 4]
```diff
@@ -75,7 +78,9 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
   override def closeSession(sessionHandle: SessionHandle): Unit = {
     _latestLogoutTime = System.currentTimeMillis()
     super.closeSession(sessionHandle)
-    operationManager.removeSparkSession(sessionHandle)
+    val session = operationManager.removeSparkSession(sessionHandle)
+    info(s"${session.sparkContext.sparkUser}'s session with $sessionHandle is closed," +
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:82]: In a Spark SQL egine , `session.sparkContext.sparkUser` is static, so looks unnecessary here

**Comment by yaooqinn** [comment]
LGTM. except for minor comment, thanks @xjzrc

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala (hunk 4 above)
Comment (xjzrc) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala]: ACK

**Comment by yaooqinn** [comment]
thanks~ merged to master

**Pull request closed without merging**